src/java/voldemort/server/niosocket/AsyncRequestHandler.java

```diff
@@ -187,10 +247,7 @@ protected void write(SelectionKey selectionKey) throws IOException {
 
         // If we don't have anything else to write, that means we're done with
         // the request! So clear the buffers (resizing if necessary).
-        if(outputStream.getBuffer().capacity() >= resizeThreshold)
-            outputStream.setBuffer(ByteBuffer.allocate(socketBufferSize));
-        else
-            outputStream.getBuffer().clear();
+        outputStream.clear();
```

Comment:
Nice suggestions for future work...

Using heuristics about 95th percentile request size would be pretty interesting and shouldn't be too hard. The buffers pool is probably not such a low hanging fruit, but also interesting for later.